tools/mcsema_disass/ida/get_cfg.py

```diff
@@ -1324,6 +1324,43 @@ def identify_program_entrypoints(func_eas):
   DEBUG_POP()
   return exported_funcs, exported_vars
 
+def find_main_in_ELF_file():
+  """Tries to automatically find the `main` function if we haven't found it
+  yet. IDA recognizes the pattern of `_start` calling `__libc_start_main` in
+  ELF binaries, where one of the parameters is the `main` function. IDA will
+  helpfully comment it as such."""
+
+  start_ea = idc.LocByName("_start")
+  if is_invalid_ea(start_ea):
+    start_ea = idc.LocByName("start")
+    if is_invalid_ea(start_ea):
+      return idc.BADADDR
+
+  for begin_ea, end_ea in idautils.Chunks(start_ea):
+    for inst_ea in Heads(begin_ea, end_ea):
+      comment = idc.GetCommentEx(inst_ea, 0)
+      if comment and "main" in comment:
```

Comment:
Honest question: would IDA ever generate a comment containing the substring "main" that was not the comment referring to `main()`?

Reply:
I have no idea. This heuristic is restricted to looking in the `start` function of ELFs though, and only if the user has told us to use `--entrypoint main`, and only if `main` wasn't found, so I'm not overly concerned about the generality of it :-P